src/app/webserver/app-pools/app-pools.service.ts

```diff
@@ -123,14 +123,19 @@ export class AppPoolsService {
         });
     }
 
-    recycle(pool: ApplicationPool) {
+    recycle(pool: ApplicationPool): Promise<any> {
         if (pool.status == Status.Stopped) {
             return this.start(pool);
         } else {
-            return this.stop(pool).then(_ => this.start(pool));
+            return this.restart(pool);
```

Comment:
The restart() of webserver.service.ts also uses the old pattern. Please update it with this new style of using await to be consistent.